cognite/client/_api/simulators/routine_revisions.py

```diff
@@ -0,0 +1,361 @@
+
+    def __call__(
+        self,
+        chunk_size: int | None = None,
+        routine_external_ids: Sequence[str] | None = None,
+        model_external_ids: Sequence[str] | None = None,
+        simulator_integration_external_ids: Sequence[str] | None = None,
+        simulator_external_ids: Sequence[str] | None = None,
+        all_versions: bool = False,
+        include_all_fields: bool = False,
+        limit: int | None = None,
+        sort: PropertySort | None = None,
+    ) -> Iterator[SimulatorRoutineRevision] | Iterator[SimulatorRoutineRevisionsList]:
+        """Iterate over simulator routine revisions
+
+        Fetches simulator routine revisions as they are iterated over, so you keep a limited number of simulator routine revisions in memory.
+
+        Args:
+            chunk_size (int | None): Number of simulator routine revisions to return in each chunk. Defaults to yielding one simulator routine revision a time.
+            routine_external_ids (Sequence[str] | None): Filter on routine external ids.
+            model_external_ids (Sequence[str] | None): Filter on model external ids.
+            simulator_integration_external_ids (Sequence[str] | None): Filter on simulator integration external ids.
+            simulator_external_ids (Sequence[str] | None): Filter on simulator external ids.
+            all_versions (bool): If all versions of the routine should be returned. Defaults to false which only returns the latest version.
+            include_all_fields (bool): If all fields should be included in the response. Defaults to false which does not include script, configuration.inputs and configuration.outputs in the response.
+            limit (int | None): Maximum number of simulator routine revisions to return. Defaults to return all items.
+            sort (PropertySort | None): The criteria to sort by.
+
+        Returns:
+            Iterator[SimulatorRoutineRevision] | Iterator[SimulatorRoutineRevisionsList]: yields SimulatorRoutineRevision one by one if chunk is not specified, else SimulatorRoutineRevisionsList objects.
+        """
+        self._warning.warn()
+        filter = SimulatorRoutineRevisionsFilter(
+            all_versions=all_versions,
+            routine_external_ids=routine_external_ids,
+            model_external_ids=model_external_ids,
+            simulator_integration_external_ids=simulator_integration_external_ids,
+            simulator_external_ids=simulator_external_ids,
+        )
+        return self._list_generator(
+            method="POST",
+            limit=limit,
+            url_path="/simulators/routines/revisions/list",
+            resource_cls=SimulatorRoutineRevision,
+            list_cls=SimulatorRoutineRevisionsList,
+            chunk_size=chunk_size,
+            filter=filter.dump(),
+            sort=[PropertySort.load(sort).dump()] if sort else None,
+            other_params={"includeAllFields": include_all_fields},
+        )
+
+    def list(
```

Comment:
Suggest moving this method to the end of the class as it overwrites the builtin `list`. I have been burned before when adding private methods that use `list` in the type hint and it has been overwritten.